services/storage/src/simcore_service_storage/s3_client.py

```diff
@@ -48,6 +51,40 @@ def from_botocore_object(obj: ObjectTypeDef) -> "S3MetaData":
         )
 
 
+async def _list_objects_v2_all_items(
+    client: S3Client,
+    bucket: S3BucketName,
+    prefix: str,
+    max_items: int | None = MAX_LIST_OBJECTS_V2_ITEMS,
+) -> list[ObjectTypeDef]:
+    """
+    Scans the given path in prefix and returns information about the files.
+    If `max_items` is None returns all the items in the dataset.
+
+    Keyword Arguments:
+        max_items -- total items to return regardless of pagination
+            (default: {MAX_LIST_OBJECTS_V2_ITEMS})
+    """
+    # NOTE: code was refactored to behave as before. There are risks
+    # of making storage crash, currently the same limits as before
+    # will apply. At most 1000 elements will be returned.
+    # Pagination will be added in the future starting from here.
+
+    pagination_config: PaginatorConfigTypeDef = {}
+    if max_items is not None:
+        pagination_config["MaxItems"] = max_items
```

Comment:
question, does AWS anyway accept more than 1000 objects here? when yes you should cap this.

Reply:
no cap just a default value